components/infra-proxy-service/server/proxy.go

```diff
@@ -45,18 +44,17 @@ func NewChefClient(config *ChefConfig) (*ChefClient, error) {
 	return &ChefClient{client: client}, nil
 }
 
-func (s *Server) createClient(ctx context.Context, orgID string) (*ChefClient, error) {
+func (s *Server) createClient(ctx context.Context, orgID string, serverID string) (*ChefClient, error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	UUID, err := uuid.FromString(orgID)
+	// TODO: combine get server & org query in one statement.
+	server, err := s.service.Storage.GetServer(ctx, serverID)
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "invalid org ID: %s", err.Error())
+		return nil, service.ParseStorageError(err, serverID, "server")
 	}
 
-	// TODO: Call GetOrgByName in order to avoid separate query to fetch infra server detail
-	// Prefer LEFT OUTER join to in order to append server detail in response
-	org, err := s.service.Storage.GetOrg(ctx, UUID)
+	org, err := s.service.Storage.GetOrg(ctx, orgID, serverID)
 	if err != nil {
 		return nil, service.ParseStorageError(err, orgID, "org")
```

Comment:
Here calling the ParseStorageError interface value with string object, so handle `getFieldValue` based on its kind working fine.